### PR TITLE
Fix Issue #490

### LINKED ISSUE #490: missing status text "Not Found" in response status line
```
extern crate actix_web;
use actix_web::{server, App, HttpResponse};

fn main() {
    server::new(|| {
        App::new()
        .prefix("/app")
        .resource("", |r| r.f(|_| HttpResponse::Ok()))  // <- handle `/app` path
        .resource("/", |r| r.f(|_| HttpResponse::Ok())) // <- handle `/app/` path
        .resource("/test", |r| {
            r.get().f(|_| HttpResponse::Ok());
            r.head().f(|_| HttpResponse::MethodNotAllowed());
        })
    }).bind("localhost:3000")
        .unwrap()
        .run();
}
```
```
$ curl -I http://localhost:3000/app/z
HTTP/1.1 404 Not Found  // <= it works here

$ curl -I http://localhost:3000/app
HTTP/1.1 200 OK

$ curl -I http://localhost:3000/ap
HTTP/1.1 404  // <= missing `Not Found` here
```

### PR DESCRIPTION
**Pull request by Neopallium**
Includes a testcase to check for the reason text in an HTTP 404 response.

The router has a default handler for 404 responses that correctly sends the "Not Found" reason in the response, because it uses the normal HTTP Response writing code.

If no handler is found (i.e. no routes), then the HTTP1 handler creates a ServerError task to handle the response writing.  ServerError wasn't including the HTTP status reason text.

Closes #490

### CODE REVIEW

File: src/server/error.rs [hunk 1]
```diff
@@ -21,7 +21,12 @@ impl HttpHandlerTask for ServerError {
             bytes.reserve(helpers::STATUS_LINE_BUF_SIZE + 1);
             helpers::write_status_line(self.0, self.1.as_u16(), bytes);
         }
+        // Convert Status Code to Reason.
+        let reason = self.1.canonical_reason().unwrap_or("<unknown status code>");
```
Comment (DoumanAsh) [inline src/server/error.rs]: Should we just leave out unknown status code?

File: tests/test_server.rs [hunk 2]
```diff
@@ -932,6 +932,29 @@ fn test_application() {
     assert!(response.status().is_success());
 }
 
+#[test]
+fn test_issue_490() {
```
Comment (DoumanAsh) [inline tests/test_server.rs]: well, I guess you could just name it `test_default_handler_response` 😄

**Review by DoumanAsh — APPROVED** [review]

**Comment by Neopallium** [comment]
I checked what nginx uses for unknown status codes.  They don't send a reason text.  So I have updated the PR to do the same.

**Comment by Neopallium** [comment]
FYI, `src/httpresponse` can return `<unknown status code>` for unknown codes.  Maybe that should be changed too.

**Comment by DoumanAsh** [comment]
@Neopallium Thanks, it would be good then to change it there too I guess, but not necessary for this PR 😄

**Pull request merged**